3rdparty/tools/rust/BUILD

```diff
@@ -0,0 +1,58 @@
+# Copyright 2023 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+file(
+    name="rustup-init",
+    source=per_platform(
+        linux_arm64=http_source(
+            url="https://static.rust-lang.org/rustup/dist/aarch64-unknown-linux-gnu/rustup-init",
```

Comment:
This looks unversioned... which is a bit concerning. What happens if rustup does a new release? 🤔 

I filed https://github.com/rust-lang/rustup/issues/3484 . Depending on the outcome there, I guess we could rehost the current binaries somewhere as a temporary (hopefully) workaround?

Reply:
If you sniff the network, it's backed by an S3 bucket.
Unfortunately https://static.rust-lang.org/?prefix=rustup/dist only yields unversioned objects,
https://static.rust-lang.org/?prefix=dist/2023-08-24/rustup comes up dry as does any hits in https://static.rust-lang.org/manifests.txt

So I think it may be unversioned publicly.

Reply:
Found it! :smile: 

https://static.rust-lang.org/?prefix=rustup/archive/

Reply:
From the repo: https://github.com/rust-lang/rustup/blob/d4c684485cb5c817d00aa6c6c0aee44da80450c7/ci/sync-dist.py#L97C72-L97C79